src/common/device_helpers.cuh

```diff
@@ -919,7 +939,9 @@ class AllReducer {
    */
   void GroupEnd() {
 #ifdef XGBOOST_USE_NCCL
-    dh::safe_nccl(ncclGroupEnd());
+    if (!use_nccl_opg) {
```

Comment:
What happens if we remove this if statement?

Reply:
> @RAMitchell @mt-jones Let me know if you need any assistance regarding the Jenkins environment.

I'm getting SSL connection failure on some Jenkins builds. Can you provide some guidance?